src/api/middleware/validators/offer.js

```diff
@@ -17,13 +17,28 @@ const create = useExpressValidators([
 
     body("publishDate", ValidationReasons.DEFAULT)
         .optional()
-        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE)
+        .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
         .toDate(),
 
     body("publishEndDate", ValidationReasons.DEFAULT)
         .exists().withMessage(ValidationReasons.REQUIRED).bail()
         .isISO8601({ strict: true }).withMessage(ValidationReasons.DATE).bail()
-        .isAfter().withMessage(ValidationReasons.DATE_EXPIRED)
+        .isAfter().withMessage(ValidationReasons.DATE_EXPIRED).bail()
+        .custom((value, { req }) => {
+            const { publishDate } = req.body;
+            // Default values and also handling if it is string or date object
+            const checkPublishDate =
+                (publishDate instanceof Date ? publishDate.toISOString() : publishDate)
+                || (new Date(Date.now())).toISOString();
+
+            if (value <= checkPublishDate) {
+                // end date is earlier than publish date, error!
+                throw new Error(ValidationReasons.MUST_BE_AFTER("publishDate"));
+            }
+
+            // Returning truthy value to indicate no error ocurred
+            return true;
+        })
```

Comment:
Looks good! However, I would suggest some var names renaming:

* `value `-> `publishEndDateCandidate`
* `publishDate` -> `publishDateRaw`
* `checkPublishDate` -> `publishDate`

In this case the if would be way clearer: `if (publishEndDate <= publishDate)`

Reply:
Done. Though of other options but these ended up being the ones I agreed the most with.